README.md

```diff
@@ -7,59 +7,661 @@
+    // function.
+    ctx.state.amountSettled = ctx.backend.getMaxValueTracker('amount_settled')
+
+    // In this type of payment channel backend, the unsecured balance we want
+    // to limit is the total amount of incoming transfers minus the sum of all
+    // the settlement transfers we've received.
+    ctx.state.maxUnsecured = opts.maxUnsecured
+
+    // use some preconfigured secret for authentication
+    ctx.state.authToken = opts.authToken
+  },
+
+  getAuthToken: (ctx) => (ctx.state.authToken),
+
+  connect: async function (ctx, opts) {
+    await Network.connectToNetwork()
+
+    // establish metadata during the connection phase
+    ctx.state.prefix = 'peer.network.' + (await Network.getChannelId())
+    ctx.state.account = await Network.getChannelSource()
+    ctx.state.peer = await Network.getChannelDestination()
+    ctx.state.info = {
+      prefix: ctx.state.prefix,
+      currencyScale: 6,
+      currencyCode: 'X??',
+      connectors: []
+    }
+  },
+
+  // Synchronous functions in order to get metadata. They won't be called until
+  // after the plugin is connected.
+  getAccount: (ctx) => (ctx.state.prefix + ctx.state.account),
+  getPeerAccount: (ctx) => (ctx.state.prefix + ctx.state.peer),
+  getInfo: (ctx) => (ctx.state.info),
+
+  handleIncomingPrepare: async function (ctx, transfer) {
+    const incoming = await ctx.transferLog.getIncomingFulfilledAndPrepared() 
+
+    // Instead of getting the best claim, we're getting the sum of all our
+    // incoming settlement transfers. This tells us how much incoming money has
+    // been secured.
+    const amountReceived = await ctx.state.incomingSettlements.getIncomingFulfilledAndPrepared()
+
+    // The peer can default on 'incoming - amountReceived', so we want to limit
+    // that amount.
+    const exceeds = new BigNumber(incoming)
+      .subtract(amountReceived)
+      .greaterThan(ctx.state.maxUnsecured)
+
+    if (exceeds) {
+      throw new NotAcceptedError(transfer.id + ' exceeds max unsecured balance')
+    }
+  },
+
+  // Even though this function is designed for creating a claim, we can
+  // very easily repurpose it to make a payment for settlement.
+  createOutgoingClaim: async function (ctx, outgoingBalance) {
+    // If a new max value is set, the maxValueTracker returns the previous max
+    // value. We tell the maxValueTracker that we're gonna pay the entire
+    // outgoingBalance we owe, and then look at the difference between the last
+    // balance and the outgoingBalance to determine how much to pay.
+    // If we've already paid out more than outgoingBalance, then it won't be the
+    // max value. The maxValueTracker will return outgoingBalance as the result,
+    // and outgoingBalance - outgoingBalance is 0. Therefore, we send no payment.
+    const lastPaid = ctx.state.amountSettled.setIfMax({ value: outgoingBalance, data: null })
+    const diff = outgoingBalance - lastPaid.value
+
+    if (!diff) {
```

Comment:
> the real lastPaid amount is higher

could you elaborate on that? lastPaid is the max value before we put the new one in. If it just returned a boolean then we wouldn't know how much money to send. If it returned the current max (after we just set it) then we couldn't determine how much money to send either.